modules/gdscript/language_server/gdscript_text_document.cpp

```diff
@@ -321,10 +321,8 @@ Variant GDScriptTextDocument::hover(const Dictionary &p_params) {
 		Array contents;
 		List<const lsp::DocumentSymbol *> list;
 		GDScriptLanguageProtocol::get_singleton()->get_workspace()->resolve_related_symbols(params, list);
-		for (const lsp::DocumentSymbol *&E : list) {
-			if (const lsp::DocumentSymbol *s = E) {
-				contents.push_back(s->render().value);
-			}
+		for (const lsp::DocumentSymbol *s : list) {
+			contents.push_back(s->render().value);
```

Comment:
I *think* the code should be equivalent?

Reply:
I don't think those are equivalent. The `=` operator returns the value of the newly assigned variable. So the original code is assigning and checking for `nullptr` at the same time. So with the new code we may be accessing an null pointer.